### PR TITLE
`jellyfish-wallet` should provide non-hd `WalletNode` by default

### PR DESCRIPTION
**Pull request by fuxingloh**
#### What kind of PR is this?:

/kind refactor

#### What this PR does / why we need it:

To allow non-HdWallet use cases and implementations as not all wallet implementations are HD, we should not enforce HD Nodes.

### CODE REVIEW

**Comment by monstrobishi** [comment]
Since we are working on naming here, I'm not sure if I understand the term non-hd wallet node, as the term HD node is very specific to HD wallets?

File: packages/jellyfish-wallet/src/wallet_account.ts [hunk 1]
```diff
@@ -85,8 +85,8 @@ export abstract class WalletAccount implements WalletHdNode {
 export interface WalletAccountProvider<T extends WalletAccount> {
 
   /**
-   * @param {WalletHdNode} hdNode of this wallet account
+   * @param {WalletNode} hdNode of this wallet account
    * @return WalletAccount
    */
-  provide: (hdNode: WalletHdNode) => T
+  provide: (hdNode: WalletNode) => T
```
Comment (ivan-zynesis) [inline packages/jellyfish-wallet/src/wallet_account.ts:91]:
```suggestion
  provide: (walletNode: WalletNode) => T
```

File: packages/jellyfish-wallet/src/wallet_account.ts [hunk 2]
```diff
@@ -11,9 +11,9 @@ import { DeFiAddress } from '@defichain/jellyfish-address'
  *
  * WalletAccount implementation uses NATIVE SEGWIT redeem script exclusively.
  */
-export abstract class WalletAccount implements WalletHdNode {
+export abstract class WalletAccount implements WalletNode {
   protected constructor (
-    private readonly hdNode: WalletHdNode,
+    private readonly hdNode: WalletNode,
```
Comment (ivan-zynesis) [inline packages/jellyfish-wallet/src/wallet_account.ts:16]:
```suggestion
    private readonly walletNode: WalletNode,
```

**Comment by monstrobishi** [comment]
https://arshbot.medium.com/hd-wallets-explained-from-high-level-to-nuts-and-bolts-9a41545f5b0

Quote:
Technically speaking, Hierarchical Deterministic Wallets are a tree structure where each **node** has an extended private and public key. Any **node** can have any number of children. Meaning a master key could regenerate 10 accounts in 10 different currencies each with a very high number of addresses. I chose 10 as an arbitrary number, the number could be as large as you’d like.

Imo it's either an HD Node or a private key pair, a node without the tree is no longer a node

**Comment by monstrobishi** [comment]
Perhaps call it `WalletEllipticPair` or `ExtendedEllipticPair`

**Comment by fuxingloh** [comment]
> Perhaps call it `WalletEllipticPair` or `ExtendedEllipticPair`

Make sense, let's call it `WalletEllipticPair`? can you make the change?

**Comment by monstrobishi** [comment]
Sure

**Review by ivan-zynesis — CHANGES_REQUESTED** [review]
Looking at the implementation again. `WalletHdNode `  do not really make sense, as how it can perform direct `getScript()` and `getAddress()`

multiple node derivation will be inside `HdWalletNodeProvider<WalletNode>`, maybe can just remove `WalletHdNode` entirely.

**Comment by fuxingloh** [comment]
looks good, let's get this merged and we can create a release to sync whale

**Review by ivan-zynesis — APPROVED** [review]

**Pull request merged**